core/templates/domain/statistics/PlaythroughIssueObjectFactorySpec.ts

```diff
@@ -65,4 +103,30 @@ describe('Playthrough Issue Object Factory', () => {
 
     expect(playthroughIssueObject.toBackendDict()).toEqual(playthroughDict);
   });
+
+  it('should throw error on invalid backend dict', () => {
+    const playthroughDict = {
+      issue_type: 'InvalidType',
+      issue_customization_args: {
+        state_name: {
+          value: 'state'
+        },
+        time_spent_in_exp_in_msecs: {
+          value: 1.2
+        }
+      },
+      playthrough_ids: [],
+      schema_version: 1,
+      is_valid: true
+    };
+
+    expect(() => {
+      // TS ignore is used because playthrough dict is assigned a invalid type
+      // to test errors.
+      // @ts-ignore
+      piof.createFromBackendDict(playthroughDict);
+    }).toThrowError(
+      'Backend dict does not match any known issue type: ' +
+      JSON.stringify(playthroughDict));
```

Comment:
Please prefer using `any` over adding `@ts-ignore`:

```suggestion
    const playthroughDict: any = {
      issue_type: 'InvalidType',
      issue_customization_args: {
        state_name: {
          value: 'state'
        },
        time_spent_in_exp_in_msecs: {
          value: 1.2
        }
      },
      playthrough_ids: [],
      schema_version: 1,
      is_valid: true
    };

    expect(() => piof.createFromBackendDict(playthroughDict))
      .toThrowError(
        'Backend dict does not match any known issue type: ' +
        JSON.stringify(playthroughDict));
```

Reply:
Why is that? I mean both ways we are bypassing the error? imo, using ts-ignore seems more sematically correct?

Reply:
Hmm, I’m not sure which we want more. /cc @seanlip WDYT?

Reply:
In this specific context, ts-ignore is better. As @nishantwrp I think it is more semantically representative of what is going on here.

Reply:
Sounds good, we can disregard this!